crates/iota/src/client_commands.rs

```diff
@@ -2916,6 +2950,49 @@ pub(crate) async fn dry_run_or_execute_or_serialize(
     }
 }
 
+async fn execute_dev_inspect(
+    context: &mut WalletContext,
+    signer: IotaAddress,
+    tx_kind: TransactionKind,
+    gas_budget: Option<u64>,
+    gas_price: u64,
+    gas_payment: Option<Vec<ObjectID>>,
+    gas_sponsor: Option<IotaAddress>,
+    skip_checks: Option<bool>,
+) -> Result<IotaClientCommandResult, anyhow::Error> {
+    let client = context.get_client().await?;
+    let gas_budget = gas_budget.map(iota_serde::BigInt::from);
+    let mut gas_objs = vec![];
+    let gas_objects = if let Some(gas_payment) = gas_payment {
+        for o in gas_payment.iter() {
+            let obj_ref = context.get_object_ref(*o).await?;
+            gas_objs.push(obj_ref);
+        }
+        Some(gas_objs)
+    } else {
+        None
+    };
```

Comment:
if `gas_payment` is `Some`, is then the length of `Vec<ObjectID>` always > 0? If not, it's kind of weird that we produce a `Some(gas_objs)` where `gas_objs` are possibly empy, no? :thinking:

Reply:
Haven't seen an obvious way how this function can be called with an empty vec (since it's private), but I guess its still good to handle this

Reply:
Done in https://github.com/iotaledger/iota/pull/5021/commits/a45b090f21808817cff964bb282615d39550f41b